service/storage_service.cc

```diff
@@ -946,17 +987,15 @@ class topology_coordinator {
             slogger.info("raft topology: new CDC generation requested");
 
             auto tmptr = get_token_metadata_ptr();
-            auto [gen_uuid, gen_mutations] = co_await prepare_new_cdc_generation_data(tmptr, guard, std::nullopt);
+            auto [gen_uuid, guard_, mutation] = co_await prepare_and_broadcast_cdc_generation_data(tmptr, std::move(guard), std::nullopt);
```

Comment:
Hm, this function operates on the `guard` variable, but you are introducing `guard_` here. I think it's slightly error prone and it would be better to reuse the `guard` variable - basically, you would like to put the new guard there.

How about std::moving the guard_ into guard after this line?

Reply:
Good idea, fixed.